src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs

```diff
@@ -378,6 +381,42 @@ protected uint UninstallMsi(string productCode, string logFile, bool ignoreDepen
             throw new InvalidOperationException($"Invalid configuration: elevated: {IsElevated}, client: {IsClient}");
         }
 
+        internal protected string GetWorkloadHistoryDirectory(string sdkFeatureBand)
+        {
+            return Path.Combine(Environment.GetEnvironmentVariable("PROGRAMDATA"), "dotnet", "workloads", sdkFeatureBand.ToString(), "history");
```

Comment:
Please use the get special folder methods to find this directory instead of the environment variable